aws/resource_aws_opsworks_chef.go

```diff
@@ -0,0 +1,343 @@
+package aws
+
+import (
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/opsworkscm"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+const EngineModel = "Single"
+const EngineVersion = "12"
+
+func resourceAwsOpsworksChef() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsOpsworksChefCreate,
+		Read:   resourceAwsOpsworksChefRead,
+		Update: resourceAwsOpsworksChefUpdate,
+		Delete: resourceAwsOpsworksChefDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			// TODO: do we want this? I think this is another "magic" thing that happens
+			// I think we can still associate a public IP later if we want? :shrug:
+			"associate_public_ip_address": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"backup_retention_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
+				// https://docs.aws.amazon.com/opsworks/latest/userguide/opscm-chef-backup.html
+				// I could swear I found this 1-30 limitation elsewhere, but I can't seem to find it now other than there
```

Comment:
Found docs [here](https://docs.aws.amazon.com/opsworks/latest/userguide/gettingstarted-opscm-create.html)
> A maximum of 30 backups are kept; when the maximum is reached, AWS OpsWorks for Chef Automate deletes the oldest backups to make room for new ones.
It's also on the console wizard